03-praticas/utils/utils.go

```diff
@@ -0,0 +1,135 @@
+package utils
+
+import (
+	"errors"
+	"strconv"
+	"strings"
+	"unicode"
+)
+
+func StringToSliceInt(numString string, hasVirgula bool) ([]int, error) {
+	numArray := []int{}
+	var numStrs []string
+
+	if hasVirgula {
+		numStrs = strings.Split(strings.TrimSpace(numString), ",")
+	} else {
+		numStrs = strings.Split(strings.TrimSpace(numString), "")
+	}
+
+	for _, numStr := range numStrs {
+		num, err := strconv.Atoi(strings.TrimSpace(numStr))
+		if err != nil {
+			return []int{}, errors.New("erro ao converter " + numStr + " para número: " + err.Error() + "\n")

```

Comment:
Es valido, igualmente en Go es buena practica usar el FMT para formatear el texto

```suggestion
return []int{}, errors.New(fmt.Sprintf("erro ao converter %s para número: %s", numStr, err.Error()))
```